### ISSUE TITLE
Custom*Forces are nonzero when lambda=0

### ISSUE THREAD

**Issue by pgrinaway** [choderalab/perses#164]
When the "initial potential components" are printed by the `NCMCEngine`, we get something like this:
```
initial potential before 'insert' : 336004.777863 kT
initial potential components:   [('HarmonicBondForce', Quantity(value=189.8220866815897, unit=kilojoule/mole)), ('CustomBondForce', Quantity(value=21316.236561565125, unit=kilojoule/mole)), ('HarmonicAngleForce', Quantity(value=109.75259884087667, unit=kilojoule/mole)), ('CustomAngleForce', Quantity(value=3027.3701748361514, unit=kilojoule/mole)), ('PeriodicTorsionForce', Quantity(value=176.6403225078547, unit=kilojoule/mole)), ('CustomTorsionForce', Quantity(value=0.0, unit=kilojoule/mole)), ('NonbondedForce', Quantity(value=837634.6902652165, unit=kilojoule/mole)), ('CustomNonbondedForce', Quantity(value=8364162186458531.0, unit=kilojoule/mole)), ('CustomNonbondedForce', Quantity(value=-434.5643798153403, unit=kilojoule/mole)), ('CustomBondForce', Quantity(value=2966343.3743730546, unit=kilojoule/mole)), ('CMMotionRemover', Quantity(value=0.0, unit=kilojoule/mole))]
```

Since `lambda` is 0 before the insertion, shouldn't those forces have no contribution (or near 0)? Or am I missing something?

**Comment by jchodera** [comment]
Yep.  That looks like a bug...

**Comment by pgrinaway** [comment]
I'm happy to dig deeper, but do you think the bug is in `alchemy`? or maybe even `OpenMM`?

**Comment by jchodera** [comment]
I thought the issue is that we don't currently specify `lambda_bonds`, `lambda_angles`, and `lambda_torsions` in [`default_functions` here](https://github.com/choderalab/perses/blob/master/perses/annihilation/ncmc_switching.py#L7-L13), but we do specify these.

Are you specifying your own functions somewhere in `perses`, instead of using the defaults?

**Comment by pgrinaway** [comment]
I don't think so. Let me double check.

**Comment by jchodera** [comment]
You're installing `alchemy` from the repo, or from `conda`?

**Comment by pgrinaway** [comment]
I think I may have found the problem

**Comment by pgrinaway** [comment]
(it was from conda previously)

**Comment by pgrinaway** [comment]
Nevermind...

**Comment by pgrinaway** [comment]
The functions are all the default (and the `lambda_x` are 0), yet the potentials are still:
```
[('HarmonicBondForce', Quantity(value=83.122650615936, unit=kilojoule/mole)), ('CustomBondForce', Quantity(value=54741.976573663705, unit=kilojoule/mole)), ('HarmonicAngleForce', Quantity(value=70.12813209107769, unit=kilojoule/mole)), ('CustomAngleForce', Quantity(value=2159.937235364632, unit=kilojoule/mole)), ('PeriodicTorsionForce', Quantity(value=141.72759925021043, unit=kilojoule/mole)), ('CustomTorsionForce', Quantity(value=220.52449959343548, unit=kilojoule/mole)), ('NonbondedForce', Quantity(value=-239.62785584431344, unit=kilojoule/mole)), ('CustomNonbondedForce', Quantity(value=3468.4350662071774, unit=kilojoule/mole)), ('CustomNonbondedForce', Quantity(value=114.74477542694993, unit=kilojoule/mole)), ('CustomBondForce', Quantity(value=-198.6360387663448, unit=kilojoule/mole)), ('CMMotionRemover', Quantity(value=0.0, unit=kilojoule/mole))]
``

**Comment by pgrinaway** [comment]
Not sure why `lambda_sterics` and `lambda_electrostatics` appear in the parameter list twice, but I assume this is unrelated.

**Comment by pgrinaway** [comment]
Also, I just installed `alchemy` from the git repo, still have the issue.

**Comment by pgrinaway** [comment]
Actually with the basic geometry system, I get this: 

`growth_stage` = 1:
[('CustomBondForce', Quantity(value=111567.10930879445, unit=kilojoule/mole)), ('CustomAngleForce', Quantity(value=nan, unit=kilojoule/mole)), ('CustomTorsionForce', Quantity(value=1917.7363999999995, unit=kilojoule/mole))]

`growth_stage` = 2:
[('CustomBondForce', Quantity(value=111567.10930879445, unit=kilojoule/mole)), ('CustomAngleForce', Quantity(value=nan, unit=kilojoule/mole)), ('CustomTorsionForce', Quantity(value=1917.7363999999995, unit=kilojoule/mole))]

Shouldn't these be different?

**Comment by jchodera** [comment]
One of those is NaN, isn't it?

**Comment by pgrinaway** [comment]
yeah, but shouldn't the others change?

**Comment by pgrinaway** [comment]
It appears they are assigned correctly.

**Comment by jchodera** [comment]
There isn't an issue with that `step(x)` function near integers, is there?

**Comment by pgrinaway** [comment]
I don't think that's it (you'd think that would make it unstable, no? I've tried a few times at various values of the global parameter, but get the same values). I think the source of the `nan` is probably that the relevant angle is not placed yet.

But (I will open another issue regarding this `nan`) that's a bit beside the point--there is no `step()` in the alchemical system, so that issue is still unexplained.

**Comment by jchodera** [comment]
Debugging now.

**Comment by pgrinaway** [comment]
If it helps, I've confirmed that for `lambda_bonds`= {0, 0.1, 0.2, 0.3} the `CustomBondForce` term does not change. I've also confirmed that the force has the appropriate `lambda_bonds` global parameter. The expression in that term is `'lambda_bonds*(K/2)*(r-r0)^2;'`

**Comment by jchodera** [comment]
Found the issue. Bond and angle alchemical parameters are not being set in context. Network is down, but will have fix up soon.

**Comment by pgrinaway** [comment]
But even when I do:

```python
context.setParameter("lambda_bonds", 0.3)
print(compute_potential_components(context))
context.setParameter("lambda_bonds", 0.0)
print(compute_potential_components(context))
```

I get
```
        
[('HarmonicBondForce', Quantity(value=25909.151349799475, unit=kilojoule/mole)), ('CustomBondForce', Quantity(value=51688.24988657913, unit=kilojoule/mole)), ('HarmonicAngleForce', Quantity(value=2097.132418816356, unit=kilojoule/mole)), ('CustomAngleForce', Quantity(value=1791.6881495196276, unit=kilojoule/mole)), ('PeriodicTorsionForce', Quantity(value=293.7551765306779, unit=kilojoule/mole)), ('CustomTorsionForce', Quantity(value=309.03183874682463, unit=kilojoule/mole)), ('NonbondedForce', Quantity(value=721.3825679001566, unit=kilojoule/mole)), ('CustomNonbondedForce', Quantity(value=8728.263976545495, unit=kilojoule/mole)), ('CustomNonbondedForce', Quantity(value=648.2688956682639, unit=kilojoule/mole)), ('CustomBondForce', Quantity(value=-1494.9687222023558, unit=kilojoule/mole)), ('CMMotionRemover', Quantity(value=0.0, unit=kilojoule/mole))]


[('HarmonicBondForce', Quantity(value=25909.151349799475, unit=kilojoule/mole)), ('CustomBondForce', Quantity(value=51688.24988657913, unit=kilojoule/mole)), ('HarmonicAngleForce', Quantity(value=2097.132418816356, unit=kilojoule/mole)), ('CustomAngleForce', Quantity(value=1791.6881495196276, unit=kilojoule/mole)), ('PeriodicTorsionForce', Quantity(value=293.7551765306779, unit=kilojoule/mole)), ('CustomTorsionForce', Quantity(value=309.03183874682463, unit=kilojoule/mole)), ('NonbondedForce', Quantity(value=721.3825679001566, unit=kilojoule/mole)), ('CustomNonbondedForce', Quantity(value=8728.263976545495, unit=kilojoule/mole)), ('CustomNonbondedForce', Quantity(value=648.2688956682639, unit=kilojoule/mole)), ('CustomBondForce', Quantity(value=-1494.9687222023558, unit=kilojoule/mole)), ('CMMotionRemover', Quantity(value=0.0, unit=kilojoule/mole))]
```

**Comment by jchodera** [comment]
The problem was in `perses.tests.utils.compute_potential_components()`.  It created a new `Context` in which `Force`s are assigned to separate force indexes, but forgot to copy the context parameters.  PR coming.

**Comment by pgrinaway** [comment]
Oh! thanks!

**Comment by jchodera** [comment]
```
available parameters: ['lambda_bonds', 'lambda_angles', 'lambda_sterics', 'lambda_electrostatics', 'lambda_electrostatics', 'lambda_sterics']
lambda_bonds
lambda_angles
lambda_sterics
lambda_electrostatics
lambda_electrostatics
lambda_sterics
initial potential before 'insert' : 539.433373 kT
initial potential components:   [('HarmonicBondForce', Quantity(value=73.26294664594303, unit=kilojoule/mole)), ('CustomBondForce', Quantity(value=0.0, unit=kilojoule/mole)), ('HarmonicAngleForce', Quantity(value=57.29333462710093, unit=kilojoule/mole)), ('CustomAngleForce', Quantity(value=0.0, unit=kilojoule/mole)), ('PeriodicTorsionForce', Quantity(value=864.7922573902418, unit=kilojoule/mole)), ('NonbondedForce', Quantity(value=350.18264663002964, unit=kilojoule/mole)), ('CustomNonbondedForce', Quantity(value=0.0, unit=kilojoule/mole)), ('CustomNonbondedForce', Quantity(value=0.0, unit=kilojoule/mole)), ('CustomBondForce', Quantity(value=0.0, unit=kilojoule/mole)), ('CMMotionRemover', Quantity(value=0.0, unit=kilojoule/mole))]
final potential before 'insert' : 139975114034930.921875 kT
final potential components: [('HarmonicBondForce', Quantity(value=348083555857345.3, unit=kilojoule/mole)), ('CustomBondForce', Quantity(value=1062213821860.2744, unit=kilojoule/mole)), ('HarmonicAngleForce', Quantity(value=268.1262245503904, unit=kilojoule/mole)), ('CustomAngleForce', Quantity(value=6421.886998439563, unit=kilojoule/mole)), ('PeriodicTorsionForce', Quantity(value=857.2639597775067, unit=kilojoule/mole)), ('NonbondedForce', Quantity(value=496.99999027381557, unit=kilojoule/mole)), ('CustomNonbondedForce', Quantity(value=1653.1724824866621, unit=kilojoule/mole)), ('CustomNonbondedForce', Quantity(value=-184.31974739120437, unit=kilojoule/mole)), ('CustomBondForce', Quantity(value=1084.333318514786, unit=kilojoule/mole)), ('CMMotionRemover', Quantity(value=0.0, unit=kilojoule/mole))]
```

**Comment by jchodera** [comment]
Fixed via #166
